src/sys/socket/addr.rs

```diff
@@ -748,6 +798,53 @@ impl SockAddr {
                 #[cfg(any(target_os = "ios", target_os = "macos"))]
                 Some(AddressFamily::System) => Some(SockAddr::SysControl(
                     SysControlAddr(*(addr as *const sys_control::sockaddr_ctl)))),
+                // FIXME: android is same ?
+                // #[cfg(any(target_os = "android", target_os = "linux"))]
+                #[cfg(target_os = "linux")]
+                Some(AddressFamily::Packet) => {
+                    let sll: *const libc::sockaddr_ll = mem::transmute(addr);
```

Comment:
This should work the same way as the other variants and not use `mem::transmute()`.

Reply:
okay